packages/blocks/src/api/registration.js

```diff
@@ -231,9 +231,23 @@ function getBlockSettingsFromMetadata( { textdomain, ...metadata } ) {
  * behavior. Once registered, the block is made available as an option to any
  * editor interface where blocks are implemented.
  *
+ * For more in-depth information on registering a custom block see the [Create a block tutorial](docs/how-to-guides/block-tutorial/README.md)
+ *
  * @param {string|Object} blockNameOrMetadata Block type name or its metadata.
  * @param {Object}        settings            Block settings.
  *
+ * @example
+ * ```js
+ * import { __ } from '@wordpress/i18n';
+ * import { registerBlockType } from '@wordpress/blocks'
+ *
+ * registerBlockType( 'namespace/block-name', {
+ *     title: __( 'My First Block' ),
+ *     edit: () => <div>{ __( 'Hello from the editor!' ) }</div>,
+ *     save: () => <div>{ __( 'Hello from the saved content!' ) }</div>,
```

Comment:
This example could be misleading, as typically `save` output can't be internationalised. 

As the saved content is static within a post it wouldn't have the desired effect, and it may cause block validation issues if the block is saved in one language and loaded in another.

It might be good to remove the `__`. Maybe just have a minimal example of using an attribute?
```js
edit: ( { attributes } ) => <div>{ attributes.content }</div>,
save: ( { attributes } ) => <div>{ attributes.content }</div>,
```

Reply:
This is very interesting. Has this been tested? If this is the case, we'll need to change other reference such as the `create-block` template as well. cc @gziolo

Reply:
The rule to follow for static blocks is that the `save` function return value needs to be determined entirely from attributes or hard-coded values, it can't use outside data of any kind. 

Checking the codebase this bug seems to already exist in the file block (it's the only one that uses `__` in a save function.)

To repro:
1. Upload a pdf to a file block
2. Publish the post
3. Change to a different language
4. Reload the post in the editor
5. Observe that the file block now has a validation error

I would recommend reproducing without the Gutenberg plugin active, as a lot of the translations are missing (at least for the dev version).

I'll make a seperate bug report for it if there isn't one already.

Reply:
Great catch @talldan with the File block and the translation used with the `aria-label` attribute 👍🏻 This is also a great example of where we could benefit from dynamic tokens that can be replaced on the server as discussed in https://github.com/WordPress/gutenberg/discussions/39831.

>  If this is the case, we'll need to change other reference such as the create-block template as well.

Yes, that is a great point. It never occurred to me that it might cause issues but now it seems so obvious 🙈

Reply:
OK cool. I'll update the example here and then do a PR to change the template the create-block package and related items.